otherlibs/dune-rpc/private/dune_rpc_private.mli

```diff
@@ -194,6 +194,7 @@ module Procedures : sig
     val ping : (unit, unit) Decl.Request.t
     val diagnostics : (unit, Diagnostic.t list) Decl.Request.t
     val shutdown : unit Decl.Notification.t
+    val format : (Promote_flag.t, unit) Decl.Request.t
```

Comment:
When would the user ever want to pass `Never` here? I can't really imagine when. How about we just make this PRC just force the promotion?

Reply:
That does make the CLI more consistent, but at the cost of making the RPC itself less coherent. If we wanted to properly support previewing, we'd want to somehow communicate what/where is being previewed via RPC to make this usable. Right now, it doesn't seem like a well thought out parameter.

FWIW, I've never used the `--preview` flag myself and I think it's barely useful. So I don't particularly care about supporting it.

You can make the final call here.

Reply:
Final call: the `--preview` flag, while useful, is going to be ignored by the RPC system.
If we think this is too restrictive, we are free to change this in a later PR.

Reasoning: communicating what's being previewed via RPC isn't actually the challenge, the challenge comes from the cache that doesn't know the difference between a preview run and an actual run, which means the promotion database is not updated if one runs `dune fmt --preview` followed by `dune fmt`.
Also I just want this PR to go through at this point.